packages/openapi-fetch/src/index.ts

```diff
@@ -38,38 +40,48 @@ export type PathsWith<Paths extends Record<string, PathItemObject>, PathnameMeth
 }[keyof Paths];
 /** Find first match of multiple keys */
 export type FilterKeys<Obj, Matchers> = { [K in keyof Obj]: K extends Matchers ? Obj[K] : never }[keyof Obj];
-/** handle "application/json", "application/vnd.api+json", "appliacation/json;charset=utf-8" and more */
-export type JSONLike = `${string}json${string}`;
+export type MediaType = `${string}/${string}`;
```

Comment:
Deprioritize `application/json` and just accept… anything

Of course, this may cause issues where a `text/plain` comes before `application/json`. But I’d personally rather have someone put forward that scenario where that happens in the real world.